sound/soc/sof/intel/hda.c

```diff
@@ -760,10 +765,14 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 	pci_free_irq_vectors(pci);
 
 	hda_dsp_stream_free(sdev);
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	snd_hdac_link_free_all(bus);
+#endif
 
 	iounmap(sdev->bar[HDA_DSP_BAR]);
 	iounmap(bus->remap_addr);
 
+	snd_hdac_ext_bus_exit(bus);
```

Comment:
missing ifdef?

Reply:
@plbossart Yes, I just checked it again. We also need #if here :)